src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReplacementService.cs

```diff
@@ -86,40 +101,53 @@ public class ReplacementService : IReplacementService
                     }
 
                     await replacementPlan.DataBlocks.ForEachAsync(plan =>
-                        ReplaceLinksForDataBlock(plan, replacementSubjectId));
+                        ReplaceLinksForDataBlock(plan, replacementPlan.ReplacementSubjectId));
                     await replacementPlan.Footnotes.ForEachAsync(plan =>
-                        ReplaceLinksForFootnote(plan, originalSubjectId, replacementSubjectId));
+                        ReplaceLinksForFootnote(plan, replacementPlan.OriginalSubjectId,
+                            replacementPlan.ReplacementSubjectId));
 
                     await _contentDbContext.SaveChangesAsync();
                     await _statisticsDbContext.SaveChangesAsync();
 
-                    return new Either<ActionResult, Unit>(Unit.Instance);
+                    // This Release Id can be found on the ReplacementFileReference
+                    var releaseId = (await _contentDbContext.ReleaseFileReferences
+                        .FindAsync(replacementReleaseFileReferenceId)).ReleaseId;
+
+                    return await RemoveSubjectAndFileFromRelease(releaseId, replacementPlan.OriginalSubjectId,
+                        originalReleaseFileReferenceId);
                 });
         }
 
-        private async Task<Either<ActionResult, Guid>> GetReleaseId(Guid originalSubjectId, Guid replacementSubjectId)
+        private async Task<Either<ActionResult, ReleaseFileReference>> CheckReleaseFileReferenceExists(Guid id)
+        {
+            return await _contentPersistenceHelper.CheckEntityExists<ReleaseFileReference>(id)
+                .OnSuccess(releaseFileReference => releaseFileReference.ReleaseFileType != ReleaseFileTypes.Data
+                    ? new Either<ActionResult, ReleaseFileReference>(
+                        ValidationActionResult(ReplacementFileTypesMustBeData))
+                    : releaseFileReference);
+        }
+
+        private async Task<Either<ActionResult, Unit>> CheckFilesAreForRelatedReleases(
+            ReleaseFileReference originalReleaseFileReference,
+            ReleaseFileReference replacementReleaseFileReference)
         {
-            // Get the latest Release referencing the original Subject 
-            var originalReleaseSubjectId = await (
-                    from releaseSubject in _statisticsDbContext.ReleaseSubject
-                    join newerVersion in _statisticsDbContext.Release on releaseSubject.ReleaseId equals newerVersion
+            // Get the latest Release referencing the original ReleaseFileReference
+            var originalReleaseId = await (
+                    from releaseFile in _contentDbContext.ReleaseFiles
```

Comment:
What's your general feeling about using this LINQ syntax? I haven't really seen it used much in the rest of the project so dunno if it's worth doing it this way from a consistency perspective. Personally it also just confuses me a little to look at it...

Reply:
I thought it made a bit more sense before because it read like SQL and in my head that's how I was imagining the data that I wanted to get from it. I've changed it now to be consistent with the other LINQ syntax that we use.

Reply:
Nice one. Looks more understandable to my beginner LINQ eyes now!